examples/mnist/train.cpp

```diff
@@ -79,10 +79,12 @@ static void train_lenet(const std::string& data_dir_path) {
 
   progress_display disp(static_cast<unsigned long>(train_images.size()));
   timer t;
-  int minibatch_size = 10;
+  int minibatch_size = 16;
   int num_epochs     = 30;
 
-  optimizer.alpha *= static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size));
+  optimizer.alpha *=
+    std::min(tiny_dnn::float_t(4),
+             static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size)));
```

Comment:
@pliptor Thank you for introducing the page. Actually, I searched about the optimization and found several informative pages such as these and also the one you introduced.
http://cs231n.github.io/optimization-1/
http://int8.io/comparison-of-optimization-techniques-stochastic-gradient-descent-momentum-adagrad-and-adadelta/

I wish I can understand basics of deep learning someday...